java/src/jmri/jmrix/AbstractThrottle.java

```diff
@@ -750,6 +750,13 @@ protected int intSpeed(float speed, int steps) {
         if (speed < 0) {
             return 1; // emergency stop
         }
+        // Subtract "half an output speed step" to prevent Math.ceil() to
+        // jump one speed step higher
+        // only if this does not make the value negative
+        if(speed > 1.0f / (2 * (steps - 1)) )
+        {
+            speed -= 1.0f / (2 * (steps - 1));
+        }
```

Comment:
For me, this seems to be the equivalent of `Math.round()`. So I suggest that you replace Math.ceil() with Math.round() and then you don't need this code.

Reply:
Yes, the only difference between this+Math.ceil() and Math.round() is for input values resulting in an output between 0 and 0.5. Which to me seemed the reason for using Math.ceil().

Anyhow, I'll try and make this more clear. Will also have a look at the failing test results. Tonight :)

Heiko

Reply:
If I'm reading this correctly, it does what `Math.round()` does. Perhaps you've confused `Math.round()` and `Math.floor()`? See the [Javadoc](https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#round-float-).

Reply:
For speed > 1.0f / (2 * (steps - 1)) it does, together with the following call to Math.ceil() (that was in the code before my PR).

For 0 < speed < 1.0f / (2 * (steps - 1)) it does not, but just uses Math.ceil() (as before).

I'll rework this PR to make it more clear (and I'll also find out why the tests fail).

Heiko